### PR TITLE
tidb-lightning: update docs of the CSV settings

### PR DESCRIPTION
**Pull request by kennytm**
### What is changed, added or deleted? (Required)

1. `mydumper.csv.separator` and `delimiter` can now accept multiple characters (v4.0+).
2. document the TOML double-quoted escape behavior because I have been asked multiple times how to use U+0001 as the separator (usually for migrating Oracle dumps because Oracle's RMAN somehow does not support RFC-4180) (v4.0+).
3. the `mydumper.filter` option was wrong (v4.0+).
4. document `mydumper.csv.terminator` (v5.2+).
5. deprecated `mydumper.csv.trim-last-separator` (v5.2+).

(**Note**: Items 1-3 apply to release-4.0, 5.0, 5.1, 5.2, master; Items 4-5 apply to release-5.2, master)

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@kennytm Could you involve one or two tech review? Thanks!

**Review by glorv — COMMENTED** [review]
LGTM

**Comment by TomShawn** [comment]
@hfxsd Please help review this PR from the writing perspective. Thanks!

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 1]
```diff
@@ -127,14 +144,26 @@ trim-last-separator = false
...
+    terminator = ",\n"
+    ```
+
 ### Non-configurable options
 
 TiDB Lightning does not support every option supported by the `LOAD DATA` statement. Some examples:
 
-* The line terminator must only be CR (`\r`), LF (`\n`) or CRLF (`\r\n`), which means `LINES TERMINATED BY` is not customizable.
 * There cannot be line prefixes (`LINES STARTING BY`).
 * The header cannot be simply skipped (`IGNORE n LINES`), it must be valid column names if present.
```
Comment (hfxsd) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:166]:
```suggestion
* The header cannot be simply skipped (`IGNORE n LINES`). It must be valid column names if present.
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 2]
```diff
@@ -150,7 +179,7 @@ strict-format = true
 Currently, a strict CSV file means every field occupies only a single line. In the other words, one of the following must be true:
```
Comment (hfxsd) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:179]:
```suggestion
Currently, a strict CSV file means every field occupies only a single line. In other words, one of the following must be true:
```

**Review by hfxsd — APPROVED** [review]

**Comment by hfxsd** [comment]
LGTM

**Comment by sunzhaoyang** [comment]
@glorv please approve this pr.

**Review by glorv — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]

**Pull request merged**